pages/apis/graphql/graphql_resource_limits.md

```diff
@@ -0,0 +1,181 @@
+      edges {                                # 1 point
+        node {                               # 500 points
+          slug                               # 0 points
+        }
+      }
+    }
+  }
+}
+```
+
+### Actual complexity
+
+The actual complexity is based on the results returned after the query execution, since the connection fields can return fewer nodes than requested. Lowering the requested complexity usually lowers the actual complexity of queries.
+
+Taking the same query used earlier, if the organization has only 10 pipelines, the actual complexity will be around 13.
+
+```graphql
+query RecentPipelineSlugs {
+  organization(slug: "organization-slug") {  # 1 point
+    pipelines(first: 500) {                  # 1 point
+      edges {                                # 1 point
+        node {                               # 10 points
+          slug                               # 0 points
+        }
+      }
+    }
+  }
+}
+```
+
+## Rate limits
+Buildkite has implemented two distinct limits to the GraphQL endpoints. These limits play a critical role in ensuring the platform operates smoothly and efficiently, while minimizing the risk of unnecessary downtime or system failures.
+
+By enforcing these limits, we can effectively manage and allocate the necessary resources for our GraphQL endpoints.
+
+### Single query limit
+
+Buildkite's API has a requested complexity limit of 50,000 for each individual query. This limit is enforced prior to query execution. The intention of this limit is to prevent users from requesting an excessive number of resources in a single query.
+
+As a best practice, we recommend breaking up queries into smaller, more manageable chunks and utilizing pagination to navigate through the resulting list rather than relying on a single large query.
+
+If the query exceeds the limit, the response will return HTTP 200 status code with the following error.
+
+```json
+{
+  "errors": [
+    {
+      "message": "Query has complexity of 251503, which exceeds max complexity of 50000"
+    }
+  ]
+}
+```
+
+### Time-based rate limit
+
+To ensure optimal performance, an organization can use up to 20,000 actual complexity points within a 5-minute period. By allowing a set number of actual complexity points, you have the flexibility to run queries of different sizes within a 5-minute window.
+
+As a best practice, we recommend utilizing client-side strategies like the following to manage time-based rate limits:
+
+- Caching to lower the number of API calls.
+- Queues to schedule API calls.
+- Pagination to only request the necessary data.
+
+If an organization exceeds the 20,000 point limit, the response will return HTTP 429 status code with the following error.
+
+```json
+{
+    "errors": [
+        {
+            "message": "Your organization has exceeded the limit of 20000 complexity points. Please try again in 187 seconds."
+        }
+    ]
+}
+```
+
+
+## Accessing limit details
+
+### Check time based limits
```

Comment:
Typo: Missing dash

```suggestion
### Check-time based limits
```

Reply:
Should the dash for the time-based instead? 😅

```suggestion
### Check time-based limits
```

Reply:
🤦🏻‍♂️ Yes, thank you!